api/v1alpha1/syncconfig_types.go

```diff
@@ -39,21 +39,23 @@ type (
 
 	// NamespaceSelector provides a way to specify targeted namespaces
 	NamespaceSelector struct {
-		// LabelSelector of namespaces to be targeted
+		// LabelSelector of namespaces to be targeted. Can be combined with MatchNames to include unlabelled namespaces.
 		LabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty"`
-		// MatchNames lists namespace names to be targeted
+		// MatchNames lists namespace names to be targeted. Each entry can be a Regex pattern. A namespace is included
+		// if at least one pattern matches. Invalid patterns will cause the sync to be cancelled and the status conditions
+		// will contain the error message.
 		MatchNames []string `json:"matchNames,omitempty"`
 	}
 
 	// SyncConfigStatus defines the observed state of SyncConfig
 	SyncConfigStatus struct {
-		// Conditions contain the states of the SyncConfig
+		// Conditions contain the states of the SyncConfig. A SyncConfig is considered Ready when at least one item has been synced.
 		Conditions []SyncConfigCondition `json:"conditions,omitempty" patchStrategy:"merge"`
```

Comment:
We can use the `metav1.Condition` instead of inventing our own.

Reply:
This is available only in K8s 1.19, we are on 1.18 and cannot upgrade due to dependency conflicts from operator sdk